dff/context_storages/update_scheme.py

```diff
@@ -0,0 +1,219 @@
+            values.update({"outlook_type": OutlookType.SLICE, "outlook": outlook})
+        else:
+            values.update({"outlook_type ": OutlookType.KEYS, "outlook": outlook})
+        return values
+
+    @validator("outlook", always=True)
+    def parse_slice_outlook(cls, value, values: dict):
+        field_name: str = values.get("field_name")
+        outlook_type: OutlookType = values.get("outlook_type")
+        if outlook_type == OutlookType.SLICE and isinstance(value, str):
+            value = value.strip("[]").split(":")
+            if len(value) != 2:
+                raise Exception("For outlook of type `slice` use colon-separated offset and limit integers.")
+            else:
+                value = [int(item) for item in [value[0] or 0, value[1] or -1]]
+        if not all([isinstance(item, int) for item in value]):
+            raise Exception(f"Outlook of field '{field_name}' contains non-integer values!")
+        return value
+
+
+class DictField(BaseSchemaField):
+    on_write: Literal[
+        FieldRule.IGNORE, FieldRule.UPDATE, FieldRule.HASH_UPDATE, FieldRule.UPDATE_ONCE
+    ] = FieldRule.UPDATE
+    outlook_type: Literal[OutlookType.KEYS] = Field(OutlookType.KEYS, const=True)
+    outlook: Union[str, List[Any]] = "[all]"
+
+
+class ValueField(BaseSchemaField):
+    on_write: Literal[
+        FieldRule.IGNORE, FieldRule.UPDATE, FieldRule.HASH_UPDATE, FieldRule.UPDATE_ONCE
+    ] = FieldRule.IGNORE
+    outlook_type: Literal[OutlookType.NONE] = Field(OutlookType.NONE, const=True)
+    outlook: Literal[None] = Field(None, const=True)
+
+
+class ExtraFields(str, Enum):
+    id = "id"
+    ext_id = "ext_id"
+    created_at = "created_at"
+    updated_at = "updated_at"
+
+
+class UpdateScheme(BaseModel):
+    id: ValueField = ValueField(name=ExtraFields.id)
+    requests: ListField = ListField(name="requests")
+    responses: ListField = ListField(name="responses")
+    labels: ListField = ListField(name="labels")
+    misc: DictField = DictField(name="misc")
+    framework_states: DictField = DictField(name="framework_states")
+    ext_id: ValueField = ValueField(name=ExtraFields.ext_id)
+    created_at: ValueField = ValueField(name=ExtraFields.created_at)
+    updated_at: ValueField = ValueField(name=ExtraFields.updated_at)
+
+    def mark_db_not_persistent(self):
+        for field, field_props in dict(self).items():
+            if field_props.on_write in (FieldRule.HASH_UPDATE, FieldRule.UPDATE_ONCE, FieldRule.APPEND):
+                field_props.on_write = FieldRule.UPDATE
+                setattr(self, field, field_props)
+
+    @staticmethod
+    def _get_update_field(dictionary_keys: Iterable, outlook: List, outlook_type: OutlookType) -> List:
+        if outlook_type == OutlookType.KEYS:
+            list_keys = sorted(list(dictionary_keys))
+            if len(list_keys) < 0:
+                return []
+            return list_keys[outlook[0] : min(outlook[1], len(list_keys))]  # noqa E203
```

Comment:
`outlook[1]` can not be max len for default?

Reply:
I don't think it would be easier or more efficieant to do.

Reply:
3 is new default